### ISSUE TITLE
`dataclass_transform` alias does not work with python version 3.11

### ISSUE THREAD

**Issue by RobertCraigie** [microsoft/pyright#3912]
**Describe the bug**

Pyright does not use the given alias for the synthesised `__init__` method if `pythonVersion` is set to `3.11`.

**To Reproduce**

```py
from pydantic import BaseModel, Field

class Model(BaseModel):
    prettyName: str = Field(alias='name')

print(Model(name='foo'))
reveal_type(Model.__init__)
```

With this `pyproject.toml` file:

```toml
[tool.pyright]
pythonVersion = "3.11"
typeCheckingMode = "strict"
```

And the third party library used, `pydantic==1.10.2`.

This is what pyright outputs:

```
test.py
  test.py:8:13 - error: No parameter named "name" (reportGeneralTypeIssues)
  test.py:9:13 - information: Type of "Model.__init__" is "(self: Model, *, prettyName: str = Field(alias='name')) -> None"
1 error, 0 warnings, 1 information 
```

It should be noted that I also encountered this error with `pythonVersion` being left unset in the config file.

**Expected behavior**

Pyright should not error and the revealed type should be:

```
(self: Model, *, name: str) -> None
```

**VS Code extension or command-line**

Command-line, `v.1.1.269`.

**Comment by erictraut** [comment]
This is a bug in pydantic's use of `dataclass_transform`. Prior to the finalization of PEP 681, we changed the name of one of the parameters from `field_descriptors` to `field_specifiers`. This was based on feedback. Pydantic is still using the old (obsolete) name in its code.

```python
@dataclass_transform(kw_only_default=True, field_descriptors=(Field, FieldInfo))
```

It needs to be updated to match the PEP. Please file a bug with the pydantic maintainers. In the meantime, you can make the change manually in your copy of pydantic sources if you want this to work.

I'll also see if there's a short-term workaround that I can make in pyright to help people who are using the current pydantic version.

**Comment by erictraut** [comment]
The maintainer of pydantic confirmed that this will be fixed in a dot release within the next several days. https://github.com/pydantic/pydantic/issues/4500
